### PR TITLE
Refactor(eos_designs): Using common method for raising duplicate detection error in python_modules

### LINKED ISSUE #2749: Refactor(eos_designs): Write a generic method in shared_utils for duplicate detection
So following discussion over chat here is the agreement to move forward:

The PR is merged as is and then two extra PRs should be opened:

1. to add negative tests for duplicate detection
2. to refactor all occurences of duplicate detection into a generic function in shared_utils and replace it

_Originally posted by @gmuloc in https://github.com/aristanetworks/ansible-avd/issues/2627#issuecomment-1517459908_
            

This requires #2748 to be merged first

### PR DESCRIPTION
**Pull request by ghost**
## Change Summary

Using common_method for raising duplicate detection error in python_modules

## Related Issue(s)

Fixes #2749

## Component(s) name

`arista.avd.eos_designs`

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/underlay/ethernet_interfaces.py [hunk 1]
```diff
@@ -125,19 +124,13 @@ def ethernet_interfaces(self) -> list | None:
...
-                raise AristaAvdError(
-                    f"Duplicate interface name {ethernet_interface['name']} found while generating ethernet_interfaces for underlay peer:"
-                    f" {ethernet_interface['peer']}, peer_interface: {ethernet_interface['peer_interface']}. Description on duplicate interface name:"
-                    f" {found_eth_interface['description']}"
-                )
+            append_if_not_duplicate(
+                list_of_dicts=ethernet_interfaces,
+                primary_key="name",
+                new_dict=ethernet_interface,
+                context="Ethernet Interfaces defined under underlay",
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_designs/python_modules/underlay/ethernet_interfaces.py]: There is not really a concept of "under underlay" for a user of AVD so we need to reword this

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/connected_endpoints/port_channel_interfaces.py [hunk 2]
```diff
@@ -203,18 +202,13 @@ def _add_port_channel_if_not_duplicate(self, candidate_port_channel_config, port
...
-            port_channel_interfaces.append(candidate_port_channel_config)
-            return
-
-        if matching_port_channel_config != candidate_port_channel_config:
-            # Found duplicate name with different generated configs
-            raise AristaAvdError(
-                f"Duplicate port-channel name {candidate_port_channel_config['name']} with conflicting configurations found while generating port-channels for"
-                " connected-endpoints or network-ports"
-            )
+        append_if_not_duplicate(
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_designs/python_modules/connected_endpoints/port_channel_interfaces.py]: do you still need the wrapper function `_add_port_channel_if_not_duplicate` ? it looks like we could remove it and just call `append_if_not_duplicate` in the main code?

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/underlay/router_bgp.py [hunk 3]
```diff
@@ -93,17 +92,13 @@ def router_bgp(self) -> dict | None:
...
-
-                    raise AristaAvdError(
-                        f"Duplicate ip_address {link['peer_ip_address']} found while generating BGP neighbor configuration for {link['peer']},"
-                        f" {link['peer_interface']}. Duplicate IP of {found_neighbor['description']}"
-                    )
+                append_if_not_duplicate(
+                    list_of_dicts=neighbors,
+                    primary_key="ip_address",
+                    new_dict=neighbor,
+                    context="Ip address defined under BGP neighbor for underlay",
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_designs/python_modules/underlay/router_bgp.py]:
```suggestion
                    context="IP address defined under BGP neighbor for underlay",
```

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/underlay/ethernet_interfaces.py (hunk 1 above)
Comment (Shivani-chourasiya) [reply ansible_collections/arista/avd/roles/eos_designs/python_modules/underlay/ethernet_interfaces.py]: fixed

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/connected_endpoints/port_channel_interfaces.py (hunk 2 above)
Comment (Shivani-chourasiya) [reply ansible_collections/arista/avd/roles/eos_designs/python_modules/connected_endpoints/port_channel_interfaces.py]: fixed

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/underlay/router_bgp.py (hunk 3 above)
Comment (Shivani-chourasiya) [reply ansible_collections/arista/avd/roles/eos_designs/python_modules/underlay/router_bgp.py]: fixed

**Review by gmuloc — APPROVED** [review]

**Review by ClausHolbechArista — APPROVED** [review]
